From e1c66397962cdfe1bb66c3f430f77906d4fb6ef9 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Sun, 9 Oct 2016 18:41:23 +0800
Subject: [PATCH 1169/5242] MLK-13303-8 ARM: imx: add i.mx6sll msl support

commit  d3d26408d536c3669a61d7723e32a49e34260987 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add i.MX6SLL MSL support, machine code reuses
i.MX6SL.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/Kconfig       |    9 +++++++++
 arch/arm/mach-imx/Makefile      |    1 +
 arch/arm/mach-imx/cpu.c         |    3 +++
 arch/arm/mach-imx/mach-imx6sl.c |    6 +++++-
 arch/arm/mach-imx/mxc.h         |    3 ++-
 5 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/Kconfig b/arch/arm/mach-imx/Kconfig
index f3839b0..351f271 100644
--- a/arch/arm/mach-imx/Kconfig
+++ b/arch/arm/mach-imx/Kconfig
@@ -600,6 +600,15 @@ comment "Cortex-A/Cortex-M asymmetric multiprocessing platforms"
 
 if ARCH_MULTI_V7 || ARM_SINGLE_ARMV7M
 
+config SOC_IMX6SLL
+	bool "i.MX6 SLL support"
+	select PINCTRL_IMX6SLL
+	select SOC_IMX6
+	select KEYBOARD_SNVS_PWRKEY
+
+	help
+	  This enables support for Freescale i.MX6 SLL processor.
+
 config SOC_VF610
 	bool "Vybrid Family VF610 support"
 	select ARM_GIC if ARCH_MULTI_V7
diff --git a/arch/arm/mach-imx/Makefile b/arch/arm/mach-imx/Makefile
index 0a62599..30f2b80 100644
--- a/arch/arm/mach-imx/Makefile
+++ b/arch/arm/mach-imx/Makefile
@@ -36,6 +36,7 @@ AFLAGS_imx6ull_low_power_idle.o :=-Wa,-march=armv7-a
 obj-$(CONFIG_SOC_IMX6ULL) += imx6ull_low_power_idle.o
 AFLAGS_imx7d_low_power_idle.o :=-Wa,-march=armv7-a
 obj-$(CONFIG_SOC_IMX7D) += cpuidle-imx7d.o imx7d_low_power_idle.o
+obj-$(CONFIG_SOC_IMX6SLL) += mach-imx6sl.o
 endif
 
 ifdef CONFIG_SND_IMX_SOC
diff --git a/arch/arm/mach-imx/cpu.c b/arch/arm/mach-imx/cpu.c
index 1702485..d5ba5ed 100644
--- a/arch/arm/mach-imx/cpu.c
+++ b/arch/arm/mach-imx/cpu.c
@@ -144,6 +144,9 @@ struct device * __init imx_soc_device_init(void)
 	case MXC_CPU_IMX7D:
 		soc_id = "i.MX7D";
 		break;
+	case MXC_CPU_IMX7ULP:
+		soc_id = "i.MX7ULP";
+		break;
 	default:
 		soc_id = "Unknown";
 	}
diff --git a/arch/arm/mach-imx/mach-imx6sl.c b/arch/arm/mach-imx/mach-imx6sl.c
index 5c2aac2..aa1f115 100644
--- a/arch/arm/mach-imx/mach-imx6sl.c
+++ b/arch/arm/mach-imx/mach-imx6sl.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2013-2015 Freescale Semiconductor, Inc.
+ * Copyright 2013-2016 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -43,6 +43,10 @@ static inline void imx6sl_fec_init(void)
 
 static void __init imx6sl_init_late(void)
 {
+	/* cpufreq and cpuidle will be enabled later for i.MX6SLL */
+	if (cpu_is_imx6sll())
+		return;
+
 	/* imx6sl reuses imx6q cpufreq driver */
 	if (IS_ENABLED(CONFIG_ARM_IMX6Q_CPUFREQ))
 		platform_device_register_simple("imx6q-cpufreq", -1, NULL, 0);
diff --git a/arch/arm/mach-imx/mxc.h b/arch/arm/mach-imx/mxc.h
index b57db4a..f285b54 100644
--- a/arch/arm/mach-imx/mxc.h
+++ b/arch/arm/mach-imx/mxc.h
@@ -99,7 +99,8 @@ static inline bool cpu_is_imx6(void)
 		__mxc_cpu_type == MXC_CPU_IMX6SL ||
 		__mxc_cpu_type == MXC_CPU_IMX6SX ||
 		__mxc_cpu_type == MXC_CPU_IMX6UL ||
-		__mxc_cpu_type == MXC_CPU_IMX6ULL;
+		__mxc_cpu_type == MXC_CPU_IMX6ULL ||
+		__mxc_cpu_type == MXC_CPU_IMX6SLL;
 }
 
 static inline bool cpu_is_imx7d(void)
-- 
1.7.9.5

